### ISSUE TITLE
Validation false error for VUID-04450

### ISSUE THREAD

**Issue by richard-lunarg** [KhronosGroup/Vulkan-ValidationLayers#2785]
**Describe the Issue**
This looks like an error in the Khronos validation layer. A developer reported the issue here originally:
https://vulkan.lunarg.com/issue/view/602578385df1127a24f3cb4b

I have verified that the VkPhysicalDevicePortabilitySubsetFeaturesKHR element mutableComparisonSamplers is indeed coming back VK_TRUE (I've tried this on NVidia, and Intel, user is on an AMD), so this is not a MoltenVK issue.

If I set the compareEnable of VkSamplerCreateInfo to VK_TRUE, it triggers this validation error.

**Valid Usage ID**
VULKAN ERROR: (VUID-VkDescriptorImageInfo-mutableComparisonSamplers-04450) Validation Error: [ VUID-VkDescriptorImageInfo-mutableComparisonSamplers-04450 ] Object 0: handle = 0x7f9418099818, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0xf467460 | vkUpdateDescriptorSets() (portability error): sampler comparison not available. The Vulkan spec states: If the [VK_KHR_portability_subset] extension is enabled, and VkPhysicalDevicePortabilitySubsetFeaturesKHR::mutableComparisonSamplers is VK_FALSE, then sampler must have been created with VkSamplerCreateInfo::compareEnable set to VK_FALSE. (https://vulkan.lunarg.com/doc/view/1.2.162.1/mac/1.2-extensions/vkspec.html#VUID-VkDescriptorImageInfo-mutableComparisonSamplers-04450)

**Environment:**
 - OS:  macOS
 - GPU: NVidia, AMD, Intel
 - SDK: 1.2.162, 1.2.170, 1.2.176.0, 1.2.176.1

**Additional context**
The portability extension is properly enabled as well, and I do call vkGetPhysicalDeviceFeatures2 to verify that mutableComparisonSamplers is VK_TRUE.

**Comment by ncesario-lunarg** [comment]
@richard-lunarg, I think I know what the problem is. Unfortunately I don't have a mac to verify the fix.

Out of curiosity, if you include the `VkPhysicalDevicePortabilitySubsetFeaturesKHR` structure in the pNext chain of `VkDeviceCreateInfo` (after populating it from vkGetPhysicalDeviceFeatures2), do you then not see the error as expected?

**Comment by richard-lunarg** [comment]
>Out of curiosity, if you include the VkPhysicalDevicePortabilitySubsetFeaturesKHR structure in 
>the pNext chain of VkDeviceCreateInfo (after populating it from vkGetPhysicalDeviceFeatures2),
>do you then not see the error as expected?

Yes, the above does eliminate the error. I presume though this is not a strict requirement for creating the device?

I can test a PR for you on my Mac's here. I have a largish Vulkan App I'm working on, and it was easy to reproduce this using it as a test.

**Comment by ncesario-lunarg** [comment]
Thanks for trying that out @richard-lunarg. Since this is a _feature_ and not a _property_, my understanding of the spec is that features are supposed to be explicitly enabled (with the exception being "extensions that turned into features" after promotion). So if a user creates a device without enabling `mutableComparisonSamplers`, the implementation can effectively "disable" that feature.

For this specific implementation (i.e., MoltenVK), We can make the query in the VL at `CreateDevice` time if the portability extension is enabled; that should solve this error, but only because it looks like MoltenVK enables this feature regardless of whether it was enabled at `CreateDevice` time or not. Maybe this is the intended behavior for all portability implementations and is documented somewhere?

@Tony-LunarG @jeremyg-lunarg @billhollings Any thoughts?
